Sources/PackageLoading/Diagnostics.swift

```diff
@@ -41,6 +41,10 @@ extension Diagnostic.Message {
 
         return .warning("ignoring duplicate product '\(product.name)'\(typeString)")
     }
+    
+    static func blamePackage(package: String) -> Diagnostic.Message {
+        .warning("The package \(package) is the source of the above warning")
+    }
 
     static func duplicateTargetDependency(dependency: String, target: String) -> Diagnostic.Message {
         .warning("invalid duplicate target dependency declaration '\(dependency)' in target '\(target)'")
```

Comment:
I think it might be better to just modify the existing `duplicateTargetDependency` to this:

```swift
static func duplicateTargetDependency(dependency: String, target: String, package: String) -> Diagnostic.Message {
    .warning("invalid duplicate target dependency declaration '\(dependency)' in target '\(target)' from package '\(package)'")
}
```

This way all the diagnostic information is delivered together in a single sentence to the user.

Reply:
+1

Reply:
Thanks for the PR!

I agree with both of the comments here.  This should logically be counted as one warning, not two.  In the future the diagnostics system should be improved so it can have additional notes and remarks attached, but with the current infrastructure, having one message would be better.

Reply:
Okay. I'll just edit the original warning :)